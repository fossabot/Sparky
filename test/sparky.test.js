const lib = require("../.dist/sparky");
const Sparky = lib.Sparky;

test("Render a component as expected",() => {
    const comp = Sparky.component(() => {
        const div = document.createElement("div");
        div.appendChild(document.createTextNode("Hola mundo"))
        return div;
    });
    expect(comp).toBeTruthy();
})