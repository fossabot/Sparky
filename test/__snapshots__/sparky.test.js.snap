// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount function Basic function 1`] = `
<body>
  <div>
    
                
    <p>
      Hola a todos
    </p>
    
            
  </div>
</body>
`;

exports[`Mount function Computed function 1`] = `
<body>
  <div>
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        24
      </span>
    </p>
    
            
  </div>
  <div>
    
                
    <p>
      Hola a todos 
    </p>
    
            
  </div>
</body>
`;

exports[`Mount function Computed w/ Events and State function 1`] = `
<body>
  <div>
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        24
      </span>
    </p>
    
            
  </div>
  <div>
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        undefined
      </span>
    </p>
    
            
  </div>
  <div
    onclick="functionScoped"
  >
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        undefined
      </span>
    </p>
    
            
  </div>
  <div>
    
                
    <p>
      Hola a todos 
    </p>
    
            
  </div>
</body>
`;

exports[`Mount function Computed w/ State function 1`] = `
<body>
  <div>
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        24
      </span>
    </p>
    
            
  </div>
  <div>
    
                
    <p>
      Hola a todos 
      <span
        class="computed"
      >
        undefined
      </span>
    </p>
    
            
  </div>
  <div>
    
                
    <p>
      Hola a todos 
    </p>
    
            
  </div>
</body>
`;
